app/routes/items/new.js

```diff
@@ -69,78 +54,98 @@ export default AuthorizeRoute.extend({
-        controller.set("length", null);
-        controller.set("width", null);
-        controller.set("height", null);
-        controller.set("selectedGrade", {
-          name: "B",
-          id: "B"
-        });
-        controller.set("imageKeys", "");
-      }
-      let codeId = controller.get("codeId");
-      if (codeId) {
-        let selected = this.get("store").peekRecord("code", codeId);
-        if (selected) {
-          let selectedSubform = selected.get("subform");
-          if (this.isSubformAllowed(selectedSubform)) {
-            controller.set("showAdditionalFields", true);
-            let details = await this.store.query(selectedSubform, {
-              distinct: "brand"
+  async manageSubformDetails() {
+    const controller = this.controller;
+    const store = this.store;
+    let codeId = controller.get("codeId");
+    if (codeId) {
+      let selected = store.peekRecord("code", codeId);
+      if (selected) {
+        let selectedSubform = selected.get("subform");
+        if (this.isSubformAllowed(selectedSubform)) {
+          controller.set("showAdditionalFields", true);
+          let details = await store.query(selectedSubform, {
+            distinct: "brand"
+          });
+          if (this.get("transitionFrom") === "code") {
+            controller.setProperties({
+              dropDownValues: {},
+              countryValue: {},
+              selected: []
             });
-            if (this.get("transitionFrom") === "code") {
-              controller.setProperties({
-                dropDownValues: {},
-                countryValue: {},
-                selected: []
-              });
-            }
-            controller.set("packageDetails", details);
-          } else {
-            controller.set("showAdditionalFields", false);
           }
+          controller.set("packageDetails", details);
+        } else {
+          controller.set("showAdditionalFields", false);
         }
       }
+    }
+  },
 
-      var imageKey = controller.get("imageKeys");
-      if (
-        imageKey &&
-        imageKey.length &&
-        window.localStorage.isSelectLocationPreviousRoute === "true"
-      ) {
-        var image = this.get("store")
-          .peekAll("image")
-          .filterBy("cloudinaryId", imageKey)
-          .get("firstObject");
-        image =
-          image ||
-          this.get("store").createRecord("image", {
-            cloudinaryId: imageKey,
-            favourite: true
-          });
-        controller.set("newUploadedImage", image);
-      } else {
-        controller.set("newUploadedImage", null);
-      }
-      window.localStorage.setItem("isSelectLocationPreviousRoute", false);
+  setUpPackageImage() {
+    const controller = this.controller;
+    const imageKey = controller.get("imageKeys");
+    const store = this.store;
+    if (
+      imageKey &&
+      imageKey.length &&
+      window.localStorage.isSelectLocationPreviousRoute === "true"
```

Comment:
Same comment as above. Why an image only when we come from select location ? I can't justify that in anyways.

Logic is too convoluted, why can't a `if (packageHasImage) -> displayimage` logic work here